src/common/utils.lua

```diff
@@ -258,82 +313,93 @@ function utils.camelCase(str)
 	return str
 end
 
+---@description gets the state of the relavent ANT globals
+---@return AntState
+function utils.getState()
+	return {
+		Records = Records,
+		Controllers = Controllers,
+		Balances = Balances,
+		Owner = Owner,
+		Name = Name,
+		Ticker = Ticker,
+		Logo = Logo,
+		Description = Description,
+		Keywords = Keywords,
+		Denomination = Denomination,
+		TotalSupply = TotalSupply,
+		Initialized = Initialized,
+		["Source-Code-TX-ID"] = SourceCodeTxId,
+	}
+end
+
 utils.notices = {}
 
-function utils.notices.credit(msg)
-	local notice = {
-		Target = msg.Recipient,
-		Action = "Credit-Notice",
-		Sender = msg.From,
-		Quantity = tostring(1),
-	}
-	for tagName, tagValue in pairs(msg) do
+--- @param oldMsg AoMessage
+--- @param newMsg AoMessage
+--- @description Add forwarded tags to the new message
+--- @return AoMessage
+function utils.notices.addForwardedTags(oldMsg, newMsg)
+	for tagName, tagValue in pairs(oldMsg) do
 		-- Tags beginning with "X-" are forwarded
 		if string.sub(tagName, 1, 2) == "X-" then
-			notice[tagName] = tagValue
+			newMsg[tagName] = tagValue
 		end
 	end
+	return newMsg
+end
 
-	return notice
+--- @param msg AoMessage
+--- @description Create a credit notice message
+--- @return AoMessage
+function utils.notices.credit(msg)
```

Comment:
nit: move these to `notices` file and out of `utils`

Reply:
https://github.com/ar-io/ar-io-ant-process/pull/32/commits/abad18f533d8b15693814d61f948dabcc1b9f240

Reply:
After moving these I realized it broke some stuff. These notice methods are used in utils and the notice methods use utils. For some reason the requires for each other caused circular references. Solved by not using the utils in the notices file - https://github.com/ar-io/ar-io-ant-process/pull/32/commits/e9c857133f4825bb4954cddd2f1ceb9ef7c433e6